ui/src/wizard/AzureWizard.ts

```diff
@@ -72,18 +75,26 @@ export class AzureWizard<T extends IInternalActionContext> implements types.Azur
                 if (step.shouldPrompt(this._context)) {
                     step.propertiesBeforePrompt = Object.keys(this._context).filter(k => !isNullOrUndefined(this._context[k]));
 
+                    const loadingQuickPick = this._showLoadingPrompt ? createQuickPick(this._context, {
+                        loadingPlaceHolder: 'Loading...'
+                    }) : undefined;
+
                     const disposable: vscode.Disposable = this._context.ui.onDidFinishPrompt((result) => {
                         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                         step!.prompted = true;
+                        loadingQuickPick?.show();
                         if (typeof result.value === 'string' && !result.matchesDefault && this.currentStepId && !step?.supportsDuplicateSteps) {
                             this._cachedInputBoxValues[this.currentStepId] = result.value;
                         }
                     });
 
                     try {
                         this.currentStepId = getEffectiveStepId(step);
+                        loadingQuickPick?.show();
                         await step.prompt(this._context);
+                        loadingQuickPick?.hide();
```

Comment:
Just do this once in in the `finally` block instead of duplicated here and in the `catch` block. Also, should probably be after `disposable.dispose()` just to be safe on timing issues